sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java

```diff
@@ -660,13 +1113,42 @@ public CarbonWriter build() throws IOException, InvalidLoadOptionException {
       // removed from the load. LoadWithoutConverter flag is going to point to the Loader Builder
       // which will skip Conversion Step.
       loadModel.setLoadWithoutConverterStep(true);
-      return new AvroCarbonWriter(loadModel, hadoopConf, this.avroSchema);
+      AvroCarbonWriter avroCarbonWriter = new AvroCarbonWriter(loadModel,
+          hadoopConf, this.avroSchema);
+      if (!StringUtils.isEmpty(filePath)) {
```

Comment:
This condition never seem to fail ?? Same for json writer case below.
We do not have similar check for parquet and orc types below

Reply:
Yeah, this is valid because we provide to write a single object also for JSON, CSV and AVRO while we don't support writing single object for PARQUET or ORC. That's why this check is not present for parquet and ORC